app/scripts/shell/dimSearchFilter.directive.js

```diff
@@ -110,15 +111,21 @@
       }
 
       filterFn = function(item) {
-        return (_.reduce(tempFns, function(memo, fn) {
-          return memo || fn(item);
-        }, false));
+        var checks = 0;
+        _.each(filters, function(filter){
+          if(item.hasOwnProperty([filter.predicate])){
+            if(item[filter.predicate]){
+              item[filter.predicate].toLowerCase() === filter.value ? checks++ : null;
+            }
+          }
+        });
+        return checks === filters.length ? true : false;
       };
 
       _.each(dimStoreService.getStores(), function(store) {
         _.chain(store.items)
           .each(function(item) {
-            item.visible = filterFn(item);
+            filters.length > 0 ? item.visible = filterFn(item) : item.visible = true;
```

Comment:
This would be easier to understand if you wrote it like this... 

item.visible = (filters.length > 0) ?  filterFn(item) : true;